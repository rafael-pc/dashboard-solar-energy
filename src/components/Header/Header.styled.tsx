import styled from "styled-components";
import { theme } from "../../styles/theme";

export const HeaderStyle = styled.div`
  width: 100%;
  display: flex;
  justify-content: left;
  align-items: center;
  padding: 20px;
  background-color: ${theme.colors.neutral_100};
  border-bottom: 3px solid ${theme.colors.bg_primary};

  h2 {
    font-size: 1.6rem;
    font-weight: ${theme.font.bolder};
    margin-left: 5%;
  }

  /* @media only screen and (max-width: 1200px) {
    padding: 15px;
    
    h2 {
      font-size: 1.3rem;
    }
  } */

  @media only screen and (max-width: 900px) {
    justify-content: center;
    padding-left: 0;
    padding: 12px;
    font-size: 0.7rem;
    text-align: center;
    line-height: 2.2rem;

    h2 {
      font-size: 1.3rem;
      margin-left: 0;
    }
  }

  @media only screen and (max-width: 641px) {
    justify-content: center;
    text-align: center;
    line-height: 2.2rem;
    padding: 5px;

    h2 {
      font-size: 1.2rem;
      margin-left: 0;
    }
  }

  @media only screen and (max-height: ${theme.screen.mobile}) {
    font-size: 0.7rem;
    padding: 5px;

    h2 {
      font-size: 1.2rem;
      margin-left: 0;
    }
  }
`;
