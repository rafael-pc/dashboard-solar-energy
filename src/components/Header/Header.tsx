import { FC } from "react";
import { HeaderStyle } from "./Header.styled";

interface IHeaderProps {
  title: string;
}

const Header: FC<IHeaderProps> = ({ title }) => {
  return (
    <HeaderStyle>
      <h2>{title}</h2>
    </HeaderStyle>
  );
};

export default Header;
