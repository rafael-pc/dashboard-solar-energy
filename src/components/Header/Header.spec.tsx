import { render, screen } from "@testing-library/react";
import Header from "./Header";
import "@testing-library/jest-dom";

describe("Header", () => {
  it("renders the title correctly", () => {
    const title = "Test Title";
    const { asFragment } = render(<Header title={title} />);
    expect(asFragment()).toMatchSnapshot();

    const headerTitle = screen.getByRole("heading", { name: title });
    expect(headerTitle).toBeInTheDocument();
  });
});
