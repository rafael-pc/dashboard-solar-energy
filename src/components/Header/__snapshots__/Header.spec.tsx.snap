// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the title correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bcPKhP fumFxo"
  >
    <h2>
      Test Title
    </h2>
  </div>
</DocumentFragment>
`;
